Honour byte order for ipv6prefix_lease.

RFC3633 says the we have to include the full delegated prefix (including the status code) in the request message.

diff --git a/networking/udhcp/d6_dhcpc.c b/networking/udhcp/d6_dhcpc.c
index ed2255e..d8b4ebf 100644
--- a/networking/udhcp/d6_dhcpc.c
+++ b/networking/udhcp/d6_dhcpc.c
@@ -333,7 +333,7 @@ static void option_to_env(const uint8_t *option, const uint8_t *option_end)
  * +-+-+-+-+-+-+-+-+
  */
 			move_from_unaligned32(v32, option + 4 + 4);
-			*new_env() = xasprintf("ipv6prefix_lease=%u", (unsigned)v32);
+			*new_env() = xasprintf("ipv6prefix_lease=%u", (unsigned)ntohl(v32));
 
 			sprint_nip6(ipv6str, option + 4 + 4 + 4 + 1);
 			*new_env() = xasprintf("ipv6prefix=%s/%u", ipv6str, (unsigned)(option[4 + 4 + 4]));
@@ -1382,6 +1382,10 @@ int udhcpc6_main(int argc UNUSED_PARAM, char **argv)
 				continue;
 			case REQUESTING:
 				if (!discover_retries || packet_num < discover_retries) {
+					if (option_mask32 & OPT_d) {
+						free(client6_data.ia_pd);
+						client6_data.ia_pd = d6_copy_option(packet.d6_options, packet_end, D6_OPT_IA_PD);
+					}
 					/* send multicast select packet */
 					send_d6_select(xid);
 					timeout = discover_timeout;
